### PR TITLE
Add SetBatch

### PR DESCRIPTION
**Pull request by DanG100**
* add BatchUpdate, BatchReplace, BatchDelete

### CODE REVIEW

File: ygnmi/ygnmi.go [hunk 1]
```diff
@@ -469,3 +470,70 @@ func Delete[T any](ctx context.Context, c *Client, q ConfigQuery[T]) (*Result, e
...
 	return responseToResult(resp), nil
 }
+
+type batchOp struct {
+	path PathStruct
+	val  interface{}
+	mode setOperation
+}
+
+// SetBatch allows multiple config operation to be applied in a single transaction.
```
[the remark this answers is not in the document]
Comment (greg-dennis) [reply ygnmi/ygnmi.go:480]: And "Set" in place of "config"?

File: ygnmi/ygnmi.go [hunk 2]
```diff
@@ -469,3 +470,70 @@ func Delete[T any](ctx context.Context, c *Client, q ConfigQuery[T]) (*Result, e
...
+	req.Prefix = &gpb.Path{
+		Target: c.target,
+	}
+	log.V(1).Info(prettySetRequest(req))
+	resp, err := c.gnmiC.Set(ctx, req)
+	log.V(1).Infof("SetResponse:\n%s", prototext.Format(resp))
+	return responseToResult(resp), err
+}
+
+// BatchUpdate stores an update operation in the ConfigBatch.
```
Comment (greg-dennis) [inline ygnmi/ygnmi.go:506]: ConfigBatch -> SetBatch
same here and below

File: ygnmi/ygnmi.go [hunk 3]
```diff
@@ -469,3 +470,70 @@ func Delete[T any](ctx context.Context, c *Client, q ConfigQuery[T]) (*Result, e
...
+		val:  setVal,
+		mode: replacePath,
+	})
+}
+
+// BatchDelete stores an update operation in the ConfigBatch.
+func BatchDelete[T any](sb *SetBatch, q ConfigQuery[T]) {
+	sb.ops = append(sb.ops, &batchOp{
+		path: q.pathStruct(),
+		val:  nil,
```
Comment (greg-dennis) [inline ygnmi/ygnmi.go:536]: drop this line?

File: ygnmi/ygnmi_test.go [hunk 4]
```diff
@@ -2226,6 +2226,114 @@ func TestDelete(t *testing.T) {
...
+	for _, tt := range tests {
+		t.Run(tt.desc, func(t *testing.T) {
+			setClient.Reset()
+			setClient.AddResponse(tt.stubResponse, tt.stubErr)
+			b := &ygnmi.SetBatch{}
+			tt.addPaths(b)
+
+			got, err := b.Set(context.Background(), client)
+			if diff := errdiff.Substring(err, tt.wantErr); diff != "" {
+				t.Fatalf("Delete() returned unexpected diff: %s", diff)
```
Comment (greg-dennis) [inline ygnmi/ygnmi_test.go:2318]: Delete()->Set() here and below

**Review by greg-dennis — CHANGES_REQUESTED** [review]

**Review by greg-dennis — APPROVED** [review]

File: ygnmi/ygnmi.go [hunk 5]
```diff
@@ -469,3 +470,69 @@ func Delete[T any](ctx context.Context, c *Client, q ConfigQuery[T]) (*Result, e
...
 	return responseToResult(resp), nil
 }
+
+type batchOp struct {
+	path PathStruct
+	val  interface{}
+	mode setOperation
+}
+
+// SetBatch allows multiple Set operations to be applied in a single transaction.
```
Comment (wenovus) [inline ygnmi/ygnmi.go:480]: nit on details:
```
// SetBatch allows multiple Set operations (Replace, Update, Delete) to be applied as part of a single Set transaction.
// Use BatchUpdate, BatchReplace, BatchDelete to add operations, and then call the Set method to send the SetRequest.
```

File: ygnmi/ygnmi_test.go [hunk 6]
```diff
@@ -2362,6 +2361,114 @@ func TestBatchGet(t *testing.T) {
...
+			ygnmi.BatchDelete(sb, root.New().Parent().Child().One().Config())
+		},
+		wantRequest: &gpb.SetRequest{
+			Prefix: &gpb.Path{
+				Target: "dut",
+			},
+			Delete: []*gpb.Path{
+				testutil.GNMIPath(t, "super-container/leaf-container-struct/uint64-leaf"),
+			},
+		},
```
Comment (wenovus) [inline ygnmi/ygnmi_test.go:2440]: maybe delete this since this I think this won't be checked?

**Review by wenovus — APPROVED** [review]

**Pull request merged**